### PR TITLE
[v2] various fixes

### PR DESCRIPTION
**Pull request by waynz0r**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | 
| License         | Apache 2.0

### CODE REVIEW

File: api/v1alpha1/istiocontrolplane_types.go [hunk 1]
```diff
@@ -172,6 +173,34 @@ func (icp *PeerIstioControlPlane) GetSpec() *IstioControlPlaneSpec {
...
+
+	for k, v := range rr.Limits {
+		r.Limits[string(k)] = &Quantity{
+			Quantity: v,
+		}
+	}
+
+	return r
+}
+
```
Comment (Laci21) [inline api/v1alpha1/istiocontrolplane_types.go]: Is it deliberate that only the `Limits` are transformed, but the `Requests` are not?

File: controllers/istiocontrolplane_controller.go [hunk 2]
```diff
@@ -892,6 +896,28 @@ func (r *IstioControlPlaneReconciler) reconcileClusterReaderSecret(ctx context.C
...
+	}
+
+	for _, imgw := range imgws.Items {
+		imgw := imgw
+		if imgw.GetSpec().GetIstioControlPlane().Name != icp.Name || imgw.GetSpec().GetIstioControlPlane().Namespace != icp.Namespace {
+			continue
+		}
+		err = util.RemoveFinalizer(ctx, r.Client, &imgw, istioMeshGatewayFinalizerID, false)
+		if err != nil {
+			return err
```
Comment (Laci21) [inline controllers/istiocontrolplane_controller.go]:
```suggestion
			return errors.WrapIf(err, "could not delete finalizer from istio mesh gateway resource")
```

File: internal/assets/manifests/istio-discovery/templates/validatingwebhookconfiguration.yaml [hunk 3]
```diff
@@ -2,11 +2,7 @@
 apiVersion: admissionregistration.k8s.io/v1
 kind: ValidatingWebhookConfiguration
 metadata:
-{{- if eq .Values.global.distribution "cisco" }}
   name: {{ include "name-with-namespaced-revision" ( dict "name" "istio-validator" "context" $) }}
-{{- else }}
-  name: istio-validator{{- if not (eq .Values.revision "") }}-{{ .Values.revision }}.{{ .Release.Namespace }}-{{ .Release.Namespace }}{{- end }}
```
Comment (Laci21) [inline internal/assets/manifests/istio-discovery/templates/validatingwebhookconfiguration.yaml]: Why is this not needed anymore for upstream?

**Review by Laci21 — APPROVED** [review]

File: internal/assets/manifests/istio-discovery/templates/validatingwebhookconfiguration.yaml (hunk 3 above)
Comment (waynz0r) [reply internal/assets/manifests/istio-discovery/templates/validatingwebhookconfiguration.yaml]: There is a `VALIDATION_WEBHOOK_CONFIG_NAME` env var that seems to be handled properly in upstream istio as well.

Comment (Laci21) [reply internal/assets/manifests/istio-discovery/templates/validatingwebhookconfiguration.yaml]: I see, thx.

**Review by Laci21 — APPROVED** [review]

File: internal/assets/manifests/istio-discovery/templates/validatingwebhookconfiguration.yaml (hunk 3 above)
Comment (waynz0r) [reply internal/assets/manifests/istio-discovery/templates/validatingwebhookconfiguration.yaml]: scratch that, it was good in 1.10, not handled properly in 1.11, so i reverted the change.

**Pull request merged**